### PR TITLE
Update script_optimized and vulkan_workflow docs with new optimization options

### PR DESCRIPTION
**Pull request by salilsdesai**
Update to the following tutorials:
- https://pytorch.org/tutorials/prototype/vulkan_workflow.html
- https://pytorch.org/tutorials/recipes/script_optimized.html

This keeps them aligned with the changes to `optimize_for_mobile` as of https://github.com/pytorch/pytorch/pull/91600

Any changes in this PR which are deletes of space at the end of lines happened automatically when I saved the files

### CODE REVIEW

**Comment by salilsdesai** [comment]
@albanD @kimishpatel @SS-JIA Please review. I'm not able to add reviewers to this PR properly.

**Review by albanD — APPROVED** [review]
SGTM

File: recipes_source/script_optimized.rst [hunk 1]
```diff
@@ -194,16 +194,23 @@ The optimized model can then be saved and deployed in mobile apps:
...
+* `ReLU and hardtanh fusion` which rewrites graph by finding ReLU/hardtanh ops and fuses them together.
 
-* Dropout removal which removes dropout nodes from this module when training is false.
+* `Dropout removal` which removes dropout nodes from this module when training is false.
 
+* `Conv packed params hoisting` which moves convolution packed params to the root module, so that the convolution structs can be deleted. This decreases model size without impacting numerics.
+
+For Vulkan backend, by default, `optimize_for_mobile` will perform the following type of optimization:
+
+* `Automatic GPU transfer` which rewrites the graph such that inputs are transferred to Vulkan backend, and outputs are transferred to CPU backend.
```
Comment (kimishpatel) [inline recipes_source/script_optimized.rst:211]:
```suggestion
* `Automatic GPU transfer` which rewrites the graph such that moving input and output data to/from gpu becomes part of the model.
```

File: recipes_source/script_optimized.rst [hunk 2]
```diff
@@ -194,16 +194,23 @@ The optimized model can then be saved and deployed in mobile apps:
 
     optimized_torchscript_model.save("optimized_torchscript_model.pth")
 
-By default, `optimize_for_mobile` will perform the following types of optimizations:
+For CPU backend, by default, `optimize_for_mobile` will perform the following types of optimizations:
```
Comment (svekars) [inline recipes_source/script_optimized.rst:197]:
```suggestion
By default, for the CPU backend, `optimize_for_mobile` performs the following types of optimizations:
```

File: prototype_source/vulkan_workflow.rst [hunk 3]
```diff
@@ -112,6 +112,10 @@ PyTorch 1.7 Vulkan backend supports only float 32bit operators. The default mode
 
 The result model can be used only on Vulkan backend as it contains specific to the Vulkan backend operators.
 
+By default, ``optimize_for_mobile`` with ``backend='vulkan'`` rewrites the graph such that inputs are transferred to Vulkan backend, and outputs are transferred to CPU backend, so the model can be run on CPU inputs and produce CPU outputs. To disable this, add the argument ``optimization_blocklist={MobileOptimizerType.VULKAN_AUTOMATIC_GPU_TRANSFER}`` to ``optimize_for_mobile``. (``MobileOptimizerType`` can be imported from ``torch.utils.mobile_optimizer``)
```
Comment (svekars) [inline prototype_source/vulkan_workflow.rst:115]:
```suggestion
By default, ``optimize_for_mobile`` with ``backend='vulkan'`` rewrites the graph so  that inputs are transferred to the Vulkan backend, and outputs are transferred to the CPU backend, therefore, the model can be run on CPU inputs and produce CPU outputs. To disable this, add the argument ``optimization_blocklist={MobileOptimizerType.VULKAN_AUTOMATIC_GPU_TRANSFER}`` to ``optimize_for_mobile``. (``MobileOptimizerType`` can be imported from ``torch.utils.mobile_optimizer``)
```

File: recipes_source/script_optimized.rst [hunk 4]
```diff
@@ -194,16 +194,23 @@ The optimized model can then be saved and deployed in mobile apps:
...
 
-* ReLU and hardtanh fusion which rewrites graph by finding ReLU/hardtanh ops and fuses them together.
+* `ReLU and hardtanh fusion` which rewrites graph by finding ReLU/hardtanh ops and fuses them together.
 
-* Dropout removal which removes dropout nodes from this module when training is false.
+* `Dropout removal` which removes dropout nodes from this module when training is false.
 
+* `Conv packed params hoisting` which moves convolution packed params to the root module, so that the convolution structs can be deleted. This decreases model size without impacting numerics.
+
+For Vulkan backend, by default, `optimize_for_mobile` will perform the following type of optimization:
```
Comment (svekars) [inline recipes_source/script_optimized.rst:209]:
```suggestion
For the Vulkan backend,`optimize_for_mobile` performs the following type of optimization:
```

**Review by svekars — COMMENTED** [review]
A couple of super minor nits.

**Review by malfet — APPROVED** [review]

File: recipes_source/script_optimized.rst (hunk 4 above)
Comment (salilsdesai) [reply recipes_source/script_optimized.rst]: Why was "by default" removed here but not for the CPU section above?

File: recipes_source/script_optimized.rst [hunk 5]
```diff
@@ -194,16 +194,23 @@ The optimized model can then be saved and deployed in mobile apps:
...
+* `Insert and fold prepacked ops` which rewrites the model graph to replace 2D convolutions and linear ops with their prepacked counterparts.
 
-* ReLU and hardtanh fusion which rewrites graph by finding ReLU/hardtanh ops and fuses them together.
+* `ReLU and hardtanh fusion` which rewrites graph by finding ReLU/hardtanh ops and fuses them together.
 
-* Dropout removal which removes dropout nodes from this module when training is false.
+* `Dropout removal` which removes dropout nodes from this module when training is false.
 
+* `Conv packed params hoisting` which moves convolution packed params to the root module, so that the convolution structs can be deleted. This decreases model size without impacting numerics.
+
```
Comment (salilsdesai) [inline recipes_source/script_optimized.rst:208]: We want to add this point here:

* `Add/ReLU fusion` which finds instances of ``relu`` ops that follow ``add`` ops and fuses them into a single ``add_relu``.

**Pull request merged**